### PR TITLE
Download Case Summary

### PR DESCRIPTION
**Pull request by proteusvacuum**
Adds a "Download Case Summary" button to the case summary. Intended to quickly download a list of all the case types / case properties. 

Would love some feedback for the data that should be included here, @snopoke 

Code Buddy @emord

### CODE REVIEW

File: corehq/apps/app_manager/urls.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from django.conf.urls import patterns, url, include
 from corehq.apps.app_manager.view_helpers import DynamicTemplateView
-from corehq.apps.app_manager.views import DownloadCCZ, AppSummaryView, AppDiffView, LanguageProfilesView
+from corehq.apps.app_manager.views import DownloadCCZ, AppSummaryView, AppDiffView, LanguageProfilesView, DownloadCaseSummaryView
```
Comment (dimagimon) [inline corehq/apps/app_manager/urls.py]: E501 line too long (129 > 115 characters)

File: corehq/apps/app_manager/views/__init__.py [hunk 2]
```diff
@@ -21,7 +21,7 @@
     download_xform,
     DownloadCCZ,
 )
-from corehq.apps.app_manager.views.app_summary import AppSummaryView
+from corehq.apps.app_manager.views.app_summary import AppSummaryView, DownloadCaseSummaryView
```
Comment (dimagimon) [inline corehq/apps/app_manager/views/__init__.py]: F401 'AppSummaryView' imported but unused
F401 'DownloadCaseSummaryView' imported but unused

File: corehq/apps/app_manager/views/app_summary.py [hunk 3]
```diff
@@ -117,3 +116,122 @@ def get_form_data(self, in_data):
...
+            tuple(CASE_SUMMARY_EXPORT_HEADER_NAMES)
+        )for case_type in case_metadata.case_types)
+
+        data = list((
+            'All Case Properties',
+            self.get_case_type_rows(case_type)
+        ) for case_type in case_metadata.case_types)
+
+        data += list((
+                case_type.name,
```
Comment (dimagimon) [inline corehq/apps/app_manager/views/app_summary.py]: E126 continuation line over-indented for hanging indent

File: corehq/apps/app_manager/views/app_summary.py [hunk 4]
```diff
@@ -117,3 +116,122 @@ def get_form_data(self, in_data):
...
+                self.get_case_property_rows(case_type, language)
+            ) for case_type in case_metadata.case_types)
```
Comment (dimagimon) [inline corehq/apps/app_manager/views/app_summary.py]: E121 continuation line under-indented for hanging indent

File: corehq/apps/app_manager/views/app_summary.py [hunk 5]
```diff
@@ -117,3 +116,121 @@ def get_form_data(self, in_data):
...
+            case_type.name,
+            tuple(CASE_SUMMARY_EXPORT_HEADER_NAMES)
+        )for case_type in case_metadata.case_types)
+
+        data = list((
+            'All Case Properties',
+            self.get_case_type_rows(case_type)
+        ) for case_type in case_metadata.case_types)
+        data += list((
+                case_type.name,
```
Comment (dimagimon) [inline corehq/apps/app_manager/views/app_summary.py]: E126 continuation line over-indented for hanging indent

**Comment by snopoke** [comment]
Looks good to me. There are some examples of docs that have been produced for apps before here: https://drive.google.com/drive/folders/0Bz-nswrLHmApeEo3Z1dpaEhfcGM?usp=sharing but I think what you've got is an excellent start and we can always iterate on it once it gets some use.

**Review by snopoke — APPROVED** [review]

**Comment by czue** [comment]
:+1: this is great

**Pull request merged**